### PR TITLE
Passenger Updates

### PR DESCRIPTION
**Pull request by JFranklin3000**
Closes [issue URL]

##### In this PR
- Added ability to update driver and passenger on vehicle and Zords
- Added checks to validate space for Drivers and Passengers.

##### Testing
- drop actors on Vehicles and assign vehicle role. 
- Validate you can't drop more than max (Zord's start with just 1 driver)

### CODE REVIEW

**Review by phildominguez — CHANGES_REQUESTED** [review]

File: module/sheet-handlers/drop-handler.mjs [hunk 1]
```diff
@@ -214,6 +249,7 @@ async function setEntryAndAddActor(droppedActor, targetActor, options) {
 
   if (["vehicle", "zord"].includes(targetActor.type)) {
     entry['vehicleRole'] = options.vehicleRole;
+
```
[the remark this answers is not in the document]
Comment (JFranklin3000) [reply module/sheet-handlers/drop-handler.mjs:252]: removed

File: module/sheets/actor-sheet.mjs [hunk 2]
```diff
@@ -456,6 +456,10 @@ export class Essence20ActorSheet extends ActorSheet {
     // Open and collapse all Item contents in container
     html.find('.header-accordion-label').click(ev => onToggleHeaderAccordion(ev, this));
 
+    html.find('.vehicleRole').change(ev => onVehicleRoleUpdate(ev, this));
+
+    html.find('.numCrew').change(ev=> onCrewNumberUpdate(ev, this));
```
[the remark this answers is not in the document]
Comment (JFranklin3000) [reply module/sheets/actor-sheet.mjs:461]: fixed

File: module/sheet-handlers/power-ranger-handler.mjs [hunk 3]
```diff
@@ -67,3 +68,96 @@ export async function onMorph(actorSheet) {
...
+
+      await actor.update ({
+        [updateString]: newRole,
+      });
+
+    }
+  }
+}
+
+export async function onCrewNumberUpdate (event, actorSheet) {
```
[the remark this answers is not in the document]
Comment (JFranklin3000) [reply module/sheet-handlers/power-ranger-handler.mjs:175]: fixed

File: module/sheet-handlers/power-ranger-handler.mjs [hunk 4]
```diff
@@ -67,3 +68,96 @@ export async function onMorph(actorSheet) {
...
+  if (newRole == 'driver') {
+    if (numberOfType < actor.system.crew.numDrivers) {
+      updateValue = true;
+    }
+  } else {
+    if (numberOfType < actor.system.crew.numPassengers) {
+      updateValue = true;
+    }
+  }
+
```
[the remark this answers is not in the document]
Comment (JFranklin3000) [reply module/sheet-handlers/power-ranger-handler.mjs:99]: removed

File: lang/en.json [hunk 5]
```diff
@@ -589,9 +589,11 @@
     "VehiclePassengers": "Passengers",
     "VehicleRole": "Role",
     "VehicleRoleDriver": "Driver",
+    "VehicleRoleError": "That position is currently maxed out.",
```
[the remark this answers is not in the document]
Comment (JFranklin3000) [reply lang/en.json:592]: changed

File: lang/en.json [hunk 6]
```diff
@@ -589,9 +589,11 @@
...
     "VehicleRoleGunner": "Gunner",
     "VehicleRolePassenger": "Passenger",
     "VehicleRoleSelect": "Select a Role for the Passenger",
+    "VehicleRoleErrorTooManyCrew": "There are currently more crew members in the vehicle then your change allows.",
```
[the remark this answers is not in the document]
Comment (JFranklin3000) [reply lang/en.json:596]: Fixed for now at a later date I might do templating.

File: module/sheet-handlers/power-ranger-handler.mjs [hunk 7]
```diff
@@ -67,3 +69,135 @@ export async function onMorph(actorSheet) {
...
+  } else {
+    const dialogResult = await Dialog.wait({
+      title: "Flip Driver and Passenger?",
+      content: "<p>Are you trying to flip a driver and a passenger?</p>",
+      buttons: {
+        yes: {
+          label: "Yes",
+        },
+        no: {
+          label: "No",
```
[the remark this answers is not in the document]
Comment (JFranklin3000) [reply module/sheet-handlers/power-ranger-handler.mjs:116]: fixed

File: module/sheet-handlers/power-ranger-handler.mjs [hunk 8]
```diff
@@ -67,3 +69,135 @@ export async function onMorph(actorSheet) {
...
+      await Dialog.wait({
+        title: "Select Who to Flip With?",
+        content: `${options}`,
+        buttons: {
+          ok: {
+            label: "Ok",
+            callback: html => _flipDriverAndPassenger(actor, key, newRole, rememberOptions(html)),
+          },
+          cancel: {
+            label: "Cancel",
```
[the remark this answers is not in the document]
Comment (JFranklin3000) [reply module/sheet-handlers/power-ranger-handler.mjs:141]: fixed

File: module/sheet-handlers/power-ranger-handler.mjs [hunk 9]
```diff
@@ -67,3 +69,135 @@ export async function onMorph(actorSheet) {
...
+    } else {
+      let options = "";
+      for (const [selectedKey,passenger] of Object.entries(actor.system.actors)) {
+        if (selectedKey != key && passenger.vehicleRole == newRole) {
+          options += `<div><input type="radio" id="${selectedKey}" value="${passenger.name}" name="flip"/><label for="${passenger.name}">${passenger.name}</label></div>`;
+        }
+      }
+
+      await Dialog.wait({
+        title: "Select Who to Flip With?",
```
[the remark this answers is not in the document]
Comment (JFranklin3000) [reply module/sheet-handlers/power-ranger-handler.mjs:133]: fixed

File: module/sheet-handlers/drop-handler.mjs [hunk 10]
```diff
@@ -190,13 +190,48 @@ async function _selectVehicleLocation(droppedActor, targetActor) {
...
+    }
+  } else {
+    if (numberOfType < targetActor.system.crew.numPassengers) {
+      allowDrop = true;
+    }
+  }
+
+  if (!allowDrop) {
+    throw new Error('There are no open spaces for this type of person.');
+  }
```
[the remark this answers is not in the document]
Comment (JFranklin3000) [reply module/sheet-handlers/drop-handler.mjs:230]: fixed

**Review by phildominguez — APPROVED** [review]

**Pull request merged**